lib/modules/datasource/index.ts

```diff
@@ -433,7 +433,8 @@ function getDigestConfig(
   config: GetDigestInputConfig
 ): DigestConfig {
   const { currentValue, currentDigest } = config;
-  const packageName = config.packageName ?? config.depName;
+  const packageName =
+    config.replacementName ?? config.packageName ?? config.depName;
```

Comment:
why this change?

Reply:
this "packageName" is used to look up digests. After this change it is able to replace images with digests, as now the correct digest for the replacement is used. Without this the depName would be replaced, but the old digest would be used as the old name would be used in lookUp